crates/executor/src/blockchain_tree/externals.rs

```diff
@@ -28,7 +28,12 @@ pub struct TreeExternals<DB, C, EF> {
 
 impl<DB, C, EF> TreeExternals<DB, C, EF> {
     /// Create new tree externals.
-    pub fn new(db: DB, consensus: C, executor_factory: EF, chain_spec: Arc<ChainSpec>) -> Self {
+    pub fn new(
+        db: Arc<DB>,
```

Comment:
rn this is only used with test db,
if we eventually use this with the real thing this will also be `Arc<DB>`, right?

Reply:
correct, this is needed for #1662